src/main/java/net/kyori/text/KeybindComponent.java

```diff
@@ -51,15 +80,104 @@ public String keybind() {
     return this.keybind;
   }
 
+  /**
+   * Sets the keybind.
+   *
+   * @param keybind the keybind
+   * @return a copy of this component
+   */
+  @Nonnull
+  public KeybindComponent keybind(@Nonnull final String keybind) {
+    return new KeybindComponent(this.children, this.color, this.obfuscated, this.bold, this.strikethrough, this.underlined, this.italic, this.clickEvent, this.hoverEvent, this.insertion, checkNotNull(keybind, "keybind"));
+  }
+
   @Nonnull
   @Override
-  public Component copy() {
-    final KeybindComponent that = new KeybindComponent(this.keybind);
-    that.mergeStyle(this);
-    for(final Component child : this.children()) {
-      that.append(child);
+  public KeybindComponent append(@Nonnull Component component) {
+    this.detectCycle(component); // detect cycle before modifying
+    final List<Component> children = new ArrayList<>(this.children.size() + 1);
+    children.addAll(this.children);
+    children.add(component);
+    return new KeybindComponent(children, this.color, this.obfuscated, this.bold, this.strikethrough, this.underlined, this.italic, this.clickEvent, this.hoverEvent, this.insertion, this.keybind);
+  }
+
+  @Nonnull
+  @Override
+  public KeybindComponent color(@Nullable TextColor color) {
+    return new KeybindComponent(this.children, color, this.obfuscated, this.bold, this.strikethrough, this.underlined, this.italic, this.clickEvent, this.hoverEvent, this.insertion, this.keybind);
+  }
+
+  @Nonnull
+  @Override
+  public KeybindComponent decoration(@Nonnull TextDecoration decoration, @Nonnull TextDecoration.State state) {
```

Comment:
This method would be simplified by both a Builder-arg constructor and usage of EnumMap.

_All in the name of scalability!_